pkg/storage/unified/search/bleve.go

```diff
@@ -554,6 +551,25 @@ func toBleveSearchRequest(req *resource.ResourceSearchRequest, access authz.Acce
 		searchrequest.Query = bleve.NewConjunctionQuery(queries...) // AND
 	}
 
+	// Can we remove this? Is access ever nil?
+	if access != nil {
+		auth, ok := claims.From(ctx)
+		if !ok {
+			return nil, resource.AsErrorResult(fmt.Errorf("missing claims"))
+		}
+		checker, err := access.Compile(ctx, auth, authz.ListRequest{
```

Comment:
We re going to need to make this work with federated queries isn't it ? 
Meaning, compile one checker per group/resource ?

Reply:
I was doing something similar [here](https://github.com/grafana/grafana/pull/99042#discussion_r1919063292). But since each index is already scoped to a group/resource this should be fine, no?

Reply:
I think it's the federated argument that is confusing me 🤔 .

Reply:
We will need something special for federated -- since we are using the composite index.  This seems worth adding a test to make sure it is getting exercised properly.  (can register a manual access checker that filters fixed thigns)